src/EntityComponentManager.cc

```diff
@@ -808,6 +808,14 @@ void EntityComponentManagerPrivate::SetRemovedComponentsMsgs(Entity &_entity,
 
   // Find the entity in the message
   auto entIter = _msg.mutable_entities()->find(_entity);
+  if (entIter == _msg.mutable_entities()->end())
```

Comment:
https://github.com/ignitionrobotics/ign-gazebo/pull/495 has a better fix actually. This is just enough to avoid crashes when running the thermal camera example world

Reply:
Oh, I see. Thanks for pointing that out. Should we try to merge #495 first then, and then we can bring those changes into this PR so that we don't need this fix?

Reply:
#495 has been merged, so can we revert this check?